### PR TITLE
[Federation] Remove unnecessary functions from develop.sh as part of …

### PR DESCRIPTION
**Pull request by madhusudancs**
This is part of a big refactor to deprecate a short-lived deploy.sh
mechanism that nobody really could use due to bugs.

```release-note
`federation/deploy/deploy.sh` was an interim solution introduced in Kubernetes v1.4 to simplify the federation control plane deployment experience. Now that we have `kubefed`, we are deprecating `deploy.sh` scripts.
```

cc @kubernetes/sig-federation-misc @irfanurrehman

### CODE REVIEW

File: federation/develop/develop.sh [hunk 1]
```diff
@@ -123,7 +115,9 @@ function get_version() {
     kube_version="${KUBE_VERSION}"
   else
     # Read the version back from the versions file if no version is given.
-    kube_version="$(jq -r '.KUBE_VERSION' ${VERSIONS_FILE})"
+    kube_version="$(cat ${VERSIONS_FILE} | python -c '\
```
Comment (nikhiljindal) [inline federation/develop/develop.sh]: I am not sure if this is the best way to do this.
Feel free to cc someone else who would know more about this

File: federation/develop/develop.sh [hunk 2]
```diff
@@ -133,33 +127,6 @@ function push() {
 
   kube::log::status "Pushing hyperkube image to the registry"
   gcloud docker -- push "${KUBE_REGISTRY}/hyperkube-amd64:${kube_version}"
-
-  # Update config after build and push, but before turning up the clusters
-  # to ensure the config has the right image version tags.
-  gen_or_update_config "${kube_version}"
```
Comment (nikhiljindal) [inline federation/develop/develop.sh]: Sorry why is this not required now?
It is deprecated but we dont want to intentionally break it :)

**Comment by nikhiljindal** [comment]
To clarify this is a cleanup PR. Does not fix a specific bug/issue, right?

File: federation/develop/develop.sh (hunk 1 above)
Comment (madhusudancs) [inline federation/develop/develop.sh]: cc @ixdy. 

This was a page taken straight out of @ixdy's book - https://github.com/kubernetes/release/pull/207

File: federation/develop/develop.sh (hunk 2 above)
Comment (madhusudancs) [inline federation/develop/develop.sh]: This function was only required to generate the config for deploying clusters, not the federation control plane. I am considering completely getting rid of the cluster deployment part.

**Review by madhusudancs — COMMENTED** [review]
@nikhiljindal this is a bug fix. This fixes the test build failures we are having right now. This is a follow up to - PR #38888

**Comment by nikhiljindal** [comment]
Thanks for the clarifications
LGTM

**Pull request merged**